packages/shared/src/broker.ts

```diff
@@ -164,3 +172,95 @@ export async function requestSwapDepositAddress(
 
   return validateResponse(chainflipNetwork, response);
 }
+
+type NonBitcoinEncodedAddress = Exclude<EncodedAddress, { Btc: unknown }>;
+type BitcoinEncodedAddress = Extract<EncodedAddress, { Btc: unknown }>;
+
+function toEncodedAddress(
+  chain: Exclude<Chain, 'Bitcoin'>,
+  address: string,
+): NonBitcoinEncodedAddress;
+function toEncodedAddress(chain: 'Bitcoin', address: string): BitcoinEncodedAddress;
+function toEncodedAddress(chain: Chain, address: string): EncodedAddress;
+function toEncodedAddress(chain: Chain, address: string): EncodedAddress {
+  switch (chain) {
+    case 'Arbitrum':
+      assert(isHex(address), 'Expected hex-encoded EVM address');
+      return { Arb: hexToBytes(address) };
+    case 'Ethereum':
+      assert(isHex(address), 'Expected hex-encoded EVM address');
+      return { Eth: hexToBytes(address) } as EncodedAddress;
+    case 'Polkadot':
+      return { Dot: isHex(address) ? hexToBytes(address) : ss58.decode(address).data };
+    case 'Solana':
+      return { Sol: isHex(address) ? hexToBytes(address) : base58.decode(address) };
+    case 'Bitcoin':
+      return { Btc: bytesToHex(new TextEncoder().encode(address)) };
+    default:
+      throw new Error(`Unsupported chain: ${chain}`);
+  }
+}
+
+const toForeignChainAddress = (
+  chain: Chain,
+  address: string,
+  network: ChainflipNetwork,
+): ForeignChainAddress => {
+  switch (chain) {
+    case 'Arbitrum':
+    case 'Ethereum':
+    case 'Polkadot':
+    case 'Solana':
+      return toEncodedAddress(chain, address);
+    case 'Bitcoin': {
+      const { type, data } = bitcoin.decodeAddress(address, network);
+      return { Btc: { [type]: data } } as ForeignChainAddress;
+    }
+    default:
+      throw new Error(`Unsupported chain: ${chain}`);
+  }
+};
+
+export const buildExtrinsicPayload = (
+  swapRequest: NewSwapRequest,
+  chainflipNetwork: ChainflipNetwork,
+): SwappingRequestSwapDepositAddressWithAffiliates => {
+  const srcAsset = getInternalAsset({ asset: swapRequest.srcAsset, chain: swapRequest.srcChain });
+  const destAsset = getInternalAsset({
+    asset: swapRequest.destAsset,
+    chain: swapRequest.destChain,
+  });
+
+  const ccmParams = transformedCcmParamsSchema('0x')
```

Comment:
what does the default value do here?

Reply:
Currently we submit the ccm params to the broker without `cfParameters` I think. I wasn't sure what happens if you add the empty value (`0x`), but it's required for when you submit the extrinsic